### DISCUSSION TITLE
Issue with executing the PnP Modern search V4 solution. kindly helps

### DISCUSSION THREAD

**Discussion by GangadharMFord** [microsoft-search/pnp-modern-search#2353] [Q&A]
![PnP Modern search exe issue](https://user-images.githubusercontent.com/111119555/189673373-0f103ef6-faeb-4b23-8ed7-8556bb91bf60.PNG)
Hi Team,

please find the below brief information about my configurations:
1. I have created a file location (C:\Code) and followed all the steps which are suggested in the MS link: https://docs.microsoft.com/en-us/sharepoint/dev/spfx/set-up-your-development-environment. 
2.Then I cloned PnP modern search solution into the same directory into the new folder (C:\Code\pnp-modern-search) then followed all the steps from the github link: https://microsoft-search.github.io/pnp-modern-search/how-to-contribute/
3. Meanwhile (this mistake) i have created helloword solution from PowerShell using (yo @microsoft/sharepoint) Powershell command prompt. 

From here when ever I run gulp serve or gulp bundle , it taking only Helloworld solution but not pnp modern solution. I 'm little stuck here is there any where should i update so that while executing it point to correct project ?

NOTE: I'm new to development but with help of here and some online videos using pnp modern search solution.

my requirement will be :  Implementing auto preview in pnp modern search hence i have changed the little code (JSON event updated on file File previewcomponent.tsx and trying to test the code / functionality. 

Can you also guide me please , if clone this project solution and do some local changes then what are the important areas to change in the current solution?  thank you in advance for help us.
![Autopreview snapshot](https://user-images.githubusercontent.com/111119555/189672710-8b74cbcd-d6cc-45d6-a1a9-f569c58430e8.PNG)

**Comment by wobba** [comment]
Seems you might be missing `git clone https://github.com/microsoft-search/pnp-modern-search.git`, and the symlink part is not needed. We should update this. The following should work if you have SPFx environment set up.

* `git clone https://github.com/microsoft-search/pnp-modern-search.git`
* `cd pnp-modern-search/search-parts`
* `npm i`
* `gulp bundle --ship`
* `gulp package-solution --ship`

**Reply by GangadharMFord** [reply]
Hi Wobba, thank you for helping us in this request.
I have followed the above steps and stuck with the below error while running gulp bundle --ship command.

Could you please suggest me what else went wring... 
![image](https://user-images.githubusercontent.com/111119555/190185723-fdc8c8d8-eaf8-442b-ab44-23259f722096.png)

**Reply by FranckyC** [reply]
Did you install all dependencies with `npm i` from the project root folder? Otherwise, you could try `npm i -g webpack@4.42.0`

**Comment by GangadharMFord** [comment]
HI Team, 

Post changing my laptop, I have installed all required components and facing the below issue. please find the snapshot of it.
Another help will be, I 'm new to development environment, kindly refer us any existing video 's or steps for configure the environment which is greatly helps.

![image](https://user-images.githubusercontent.com/111119555/192208380-49de873e-d40d-4815-b54a-3fc500f8f12e.png)

**Reply by wobba** [reply]
First follow the guide at https://learn.microsoft.com/en-us/sharepoint/dev/spfx/set-up-your-development-environment, then you should be able to build these web parts.
